### DISCUSSION TITLE
Full compare and diffs of 100's MB text files with millions of lines

### DISCUSSION THREAD

**Discussion by rnsc5jjjjj** [WinMerge/winmerge#2036] [Q&A] [answered]
I have a large hierarchy of files of large text files (Currently up to many hundreds of MB) that I need to validate after changes to a system.  The goal is to brush aside the 99.9% that are correct and make it easy to see the 0.1% that needs to be examined.  

How can I configure WinMerge to simply plow through a 100% compare and line by line display of what is different, just like it does on more reasonably sized hierarchy of text files?  I have set the quick compare limit and binary compare limit to 2000 MB so that it doesn't just quit early and assume binary, but it just says "Text files are different" and does not show me the line by line differences.  It turns out that there are two or three trivial changes in the first ten lines, the rest matching (Which is expected given the changes made to the system).  The beauty of WinMerge is that if those few changes are at the end, or disbursed throughout, I can easily see them and jump between them.  If there is a bigger mess, at least I can quickly identify a place to dig into!  (I have done this on a much smaller scale for a long time).

I have 64GB of RAM and a fast SSD, so though patience will likely be required, memory should not be a problem since I have 64 times the largest file size.

I just need to change WinMerge's concept of what is reasonable.  Any ideas or insights will be greatly appreciated!

**Comment by sdottaka** [comment, marked as answer]
WinMerge asks the following questions when comparing large files:
![image](https://github.com/WinMerge/winmerge/assets/98126/584deae0-a02c-454b-a91b-306b5337cc5e)

Click the No button to display not only the results but also the contents of the file.
If this message box is not displayed, uncheck the message in the Message Boxes category of the Options window as shown below.

![image](https://github.com/WinMerge/winmerge/assets/98126/57b9fe22-79e2-4a80-a13c-a2cd88873b0f)

**Comment by rnsc5jjjjj** [comment]
Thank you!

Initially I did not have the "Message Boxes" option because I was on 
2.16.10.  It is there in 2.16.32.

Then I had to find the location pane (I was used to it being on the 
right).

In searching the options I learned alot - I have used WinMerge very 
happily for about ten years, but I found many important options that 
make it much better to be able to choose!

My compare took 4 minutes, and I have all of the information I need, 
easy to check and find.

Thank you SO much!

On 2023-09-18 18:19, Takashi Sawanaka wrote:

> WinMerge asks the following questions when comparing large files:
> [1]
> 
> Click the No button to display not only the results but also the 
> contents of the file.
> If this message box is not displayed, uncheck the message in the 
> Message Boxes category of the Options window as shown below.
> 
> [2]
> 
> --
> Reply to this email directly, view it on GitHub [3], or unsubscribe 
> [4].
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>



Links:
------
[1] 
https://user-images.githubusercontent.com/98126/268786222-584deae0-a02c-454b-a91b-306b5337cc5e.png
[2] 
https://user-images.githubusercontent.com/98126/268786468-57b9fe22-79e2-4a80-a13c-a2cd88873b0f.png
[3] 
https://github.com/WinMerge/winmerge/discussions/2036#discussioncomment-7039658
[4]
